File: Sort/1465. Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.py
class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:
        horizontalCuts.sort()
        verticalCuts.sort()
        res1, res2 = horizontalCuts[0], verticalCuts[0]
        for i in range(len(horizontalCuts)):
            if i < len(horizontalCuts)-1:
                res1 = max(res1, horizontalCuts[i+1] - horizontalCuts[i])
            else:
                 res1 = max(res1, h - horizontalCuts[i])
            
        for i in range(len(verticalCuts)):
            if i < len(verticalCuts)-1:
                res2 = max(res2, verticalCuts[i+1] - verticalCuts[i])
            else:
                res2 = max(res2, w - verticalCuts[i]) 
        
        return res1 * res2 % (10**9 + 7)

class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:
        horizontalStrips = [0] + sorted(horizontalCuts) + [h]
        verticalStrips = [0] + sorted(verticalCuts) + [w]
        
        maxStripWidth = max([horizontalStrips[i + 1] - horizontalStrips[i] for i in range(len(horizontalStrips) - 1)])
        maxStripHeight = max([verticalStrips[i + 1] - verticalStrips[i] for i in range(len(verticalStrips) - 1)])
        
        return (maxStripWidth * maxStripHeight) % ((10 ** 9) + 7)

File: Sort/1465. Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.cpp
class Solution {
    int getMax(int hh, vector<int> &h) {
        sort(h.begin(), h.end()); 
        int nh = h.size();
        int maxHeight = max(hh-h[nh-1], h[0]);
        for (int i=1; i<nh; i++) {
            maxHeight = max(maxHeight, h[i]-h[i-1]);
        }
        return maxHeight;
    }
public:
    int maxArea(int hh, int ww, vector<int>& h, vector<int>& v) {
        return (int)((long)getMax(hh, h)*getMax(ww, v)%1000000007);
    }
};

class Solution {
public:
    int maxArea(int h, int w, vector<int>& hCuts, vector<int>& vCuts) {
        sort(begin(hCuts), end(hCuts));
        sort(begin(vCuts), end(vCuts));
        auto max_h = max(hCuts[0], h - hCuts.back());
        auto max_v = max(vCuts[0], w - vCuts.back());
        for (auto i = 0; i < hCuts.size() - 1; ++i)
            max_h = max(max_h, hCuts[i + 1] - hCuts[i]);
        for (auto i = 0; i < vCuts.size() - 1; ++i)
            max_v = max(max_v, vCuts[i + 1] - vCuts[i]);        
        return (long)max_h * max_v % 1000000007;
    }
};